File: src/main/java/com/scottscmo/song/Verse.java
package com.scottscmo.song;

import java.util.Map;
import java.util.stream.Collectors;

public record Verse(
    String verse,
    Map<String, String> text
) {
    public Map<String, String[]> getText() {
        return this.text.entrySet().stream()
            .collect(Collectors.toMap(
                e -> e.getKey(),
                e -> e.getValue().split("\n")
            ));
    }
}

File: src/main/kotlin/com/scottscmo/song/Verse.kt
package com.scottscmo.song

data class Verse(val verse: String, val text: Map<String, String>) {
    fun textLines(): Map<String, List<String>> {
        return this.text.map {
            it.key to it.value.split("\n")
        }.toMap()
    }
}
